core/iwasm/compilation/aot_compiler.h

```diff
@@ -217,6 +225,15 @@ check_type_compatible(uint8 src_type, uint8 dst_type)
 #define PUSH_COND(v) PUSH(v, VALUE_TYPE_I1)
 #define PUSH_FUNCREF(v) PUSH(v, VALUE_TYPE_FUNCREF)
 #define PUSH_EXTERNREF(v) PUSH(v, VALUE_TYPE_EXTERNREF)
+#if WASM_ENABLE_GC != 0
+#define PUSH_REF(v) PUSH(v, VALUE_TYPE_GC_REF)
+#else
+#define PUSH_REF(v)                                                       \
+    do {                                                                  \
+        LOG_DEBUG("enable_gc=true when CMake flag are not set properly"); \
+        assert(0);                                                        \
```

Comment:
How about bh_assert(!"should not PUSH_REF when WASM_ENABLE_GC macro isn't enabled");